employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/pages/shiftroster/ShiftEditForm.java

```diff
@@ -253,13 +203,40 @@ public void onApplyButtonClick(@ForEvent("click") final MouseEvent e) {
     public void onDeleteButtonClick(@ForEvent("click") final MouseEvent e) {
         final ShiftView shiftView = shiftGridObject.getShiftView();
 
-        ShiftRestServiceBuilder.removeShift(shiftView.getTenantId(), shiftView.getId(), FailureShownRestCallback.onSuccess(v -> {
+        ShiftRestServiceBuilder.removeShift(shiftView.getTenantId(), shiftView.getId(), callbackFactory.onSuccess(v -> {
             shiftGridObject.getLane().removeGridObject(shiftGridObject);
-            formPopup.hide();
+            hide();
         }));
         e.stopPropagation();
     }
 
+    public boolean updateShiftFromWidgetsIfValid(ShiftView shiftView) {
+        final boolean oldLockedByUser = shiftView.isPinnedByUser();
+        final Long oldEmployee = shiftView.getEmployeeId();
+        final Long oldSpot = shiftView.getSpotId();
+        final LocalDateTime oldStartDateTime = shiftView.getStartDateTime();
+        final LocalDateTime oldEndDateTime = shiftView.getEndDateTime();
+
+        try {
+            shiftView.setPinnedByUser(pinned.checked);
+            shiftView.setSpotId(parseId(spotSelect.getSelectedValue()));
+            shiftView.setEmployeeId(parseId(employeeSelect.getSelectedValue()));
+            shiftView.setStartDateTime(from.getValue());
+            shiftView.setEndDateTime(to.getValue());
+            if (!shiftView.getStartDateTime().isBefore(shiftView.getEndDateTime())) {
+                throw new ValidationException("Shift's end time need to be after shift's start time.");
```

Comment:
The `ValidationException` never gets to the user - this method catches it, returns `false`, which is never checked. As a result, the Shift returns to the old state and user gets no information about what went wrong.

The validation on individual fields is shown on the screen - it's probably compiled into the JS. Thus I am not sure if the ValidationException needs to be caught here.

Reply:
The default error catcher doesn't work well here - it seems the `ValidationException` is converted to a Javascript exception (which in turn doesn't have a message). However, using NotificationFactory works as expected (although there the question of if we should allow the user to see the stack trace since this is an "expected" exception).